### PR TITLE
[SPOOL-420] Get tests running consistently in both travis and the docker dev env

### PR DESCRIPTION
**Pull request by marcparadise**
This builds atop on @smith's added .travis.yml and gets for the tests running in both travis and docker.

This also cleans up a few things: 
* minor unit test changes to work correctly in docker and travis env 
* include rebar 3.2
* remove override build settings for erlzmq since chef/erlzmq is (or will shortly be) updated with rebar3 support.

### CODE REVIEW

File: pushy-server-schema/Makefile [hunk 1]
```diff
@@ -8,14 +8,17 @@ setup_schema:
 	@echo "Dropping and recreating database '$(TEST_DB)'"
 	@psql --dbname template1 --single-transaction --command 'DROP DATABASE IF EXISTS $(TEST_DB)'
 	@createdb $(TEST_DB)
-#	@psql --dbname $(TEST_DB) --single-transaction -f /Users/cm/oc/code/opscode/opscode-pushy-server/db/pgsql_schema.sql
 	@sqitch --engine pg --db-name $(TEST_DB) deploy
 
 setup_tests:
-	@psql --dbname $(TEST_DB) --command 'CREATE EXTENSION pgtap;'
-	@psql --dbname $(TEST_DB) --command 'CREATE EXTENSION chef_pgtap;'
+	#@psql --dbname $(TEST_DB) --command 'CREATE EXTENSION pgtap;'
```
Comment (marcparadise) [inline pushy-server-schema/Makefile]: Need to figure out what happened here - this was working fine in local docker runs until suddenly it wasn't.

File: rebar.config [hunk 2]
```diff
@@ -10,19 +10,19 @@
   {meck,
    {git, "https://github.com/eproxus/meck", {branch, "master"}}},
   {erlzmq,
-   {git, "https://github.com/chef/erlzmq2", {branch, "devel"}}},
+   {git, "https://github.com/chef/erlzmq2", {branch, "mp/devel-rebase" }}},
```
Comment (marcparadise) [inline rebar.config]: Reminder to not forget this.

File: .gitignore [hunk 3]
```diff
@@ -21,3 +22,6 @@ log
 .bash_history
```
Comment (smith) [inline .gitignore]: We can remove these if we're going to use /srv.

File: Dockerfile [hunk 4]
```diff
@@ -1,5 +1,29 @@
...
+#  {"linux", "DRV_LDFLAGS", "-Wl,-rpath=${ZEROMQ_PREFIX}/lib $DRV_LDFLAGS"},
+# A check of the final rpath as set in the container-built binary shows:
+
+# root@5ad413cc8d5c:/srv# objdump -x  _build/default/lib/erlzmq/priv/erlzmq_drv.so | grep PATH[
+#  RPATH                /lib:/srv/deps/local/lib
+
+#
+# So the below works for now in conjunction with 'ldconfig' - but it would
+# be nice to come back and hunt it down...
+RUN echo /srv/_build/default/lib/erlzmq/deps/local/lib > /etc/ld.so.conf.d/erlzmq_drv.conf
```
Comment (smith) [inline Dockerfile]: Can we set the `LD_RUN_PATH`?

File: Makefile [hunk 5]
```diff
@@ -24,5 +24,11 @@ omnibus: install
 distclean:
 	@rm -rf _build
 
-shell:
-	if [ -z $$(docker images -q $(DOCKER_IMAGE_TAG)) ]; then docker build --tag $(DOCKER_IMAGE_TAG) .; fi; docker run --volume $(PWD):/root --interactive --tty $(DOCKER_IMAGE_TAG)
+update-image:
+	@docker build --tag $(DOCKER_IMAGE_TAG) .
```
Comment (marcparadise) [inline Makefile]: Yes, but this only adds about a second (at least in local testing, may be more on a docker mac setup?)

Comment (smith) [inline Makefile]: It seems like a waste of time. Why not:

```
@[[ ! -z $$(docker images -q $(DOCKER_IMAGE_TAG)) ]] || docker build --tag $(DOCKER_IMAGE_TAG) .
```

**Comment by marcparadise** [comment]
http://wilson.ci.chef.co/job/opscode-push-jobs-server-trigger-ad_hoc/24/

**Pull request merged**